src/include/execution/sql/index_iterator.h

```diff
@@ -0,0 +1,108 @@
+#pragma once
+
+#include <memory>
+#include <vector>
+#include "catalog/catalog_defs.h"
+#include "execution/exec/execution_context.h"
+#include "execution/sql/projected_columns_iterator.h"
+#include "storage/storage_defs.h"
+
+namespace terrier::execution::sql {
+/**
+ * Allows iteration for indices from TPL.
+ */
+class IndexIterator {
+ public:
+  /**
+   * Constructor
+   * @param table_oid oid of the table
+   * @param index_oid oid of the index to iterate over.
+   * @param exec_ctx execution containing of this query
+   * @param col_oids oids of the table columns
+   * @param num_oids number of oids
+   */
+  explicit IndexIterator(uint32_t table_oid, uint32_t index_oid, exec::ExecutionContext *exec_ctx, u32 *col_oids,
+                         u32 num_oids);
+
+  /**
+   * Initialize the projected row and begin scanning.
+   */
+  void Init();
+
+  /**
+   * Frees allocated resources.
+   */
+  ~IndexIterator();
+
+  /**
+   * Wrapper around the index's ScanKey
```

Comment:
Ditto, don't expose implementation details. Instead, give a high-level overview of what this is supposed to do to someone who doesn't know indexes or what `ScanKey` is.